packages/client/src/schema.ts

```diff
@@ -385,25 +409,119 @@ class SchemaBuilder {
     return field.required ? new GraphQLNonNull(type) : type
   }
 
-  _buildDocumentObjectViewField(
-    definitions: SharedDefinitions,
-    field: RuntimeViewField
+  _buildDocumentObjectRelation(
+    key: string,
+    relation: RuntimeRelation,
+    objectFields: RuntimeObjectFields
   ): GraphQLFieldConfig<ModelInstanceDocument, Context> {
-    if (field.viewType === 'documentAccount') {
-      return {
-        type: new GraphQLNonNull(definitions.accountObject),
-        resolve: (doc): string => doc.metadata.controller,
-      }
+    const modelAlias = this.#modelAliases[relation.model]
+    if (modelAlias == null) {
+      throw new Error(
+        `Model alias not found for relation with ID ${relation.model} on field ${key}`
+      )
     }
-    if (field.viewType === 'documentVersion') {
-      return {
-        type: new GraphQLNonNull(CeramicCommitID),
-        resolve: (doc): string => doc.commitId.toString(),
+
+    switch (relation.source) {
+      case 'document': {
+        const ref = objectFields[relation.property]
+        if (ref == null) {
+          throw new Error(
+            `Missing reference field ${relation.property} for relation defined on field ${key}`
+          )
+        }
+        return {
+          type: this.#types[modelAlias],
+          resolve: async (doc, _args, ctx): Promise<ModelInstanceDocument | null> => {
+            const id = doc.content?.[relation.property] as string | void
+            if (id == null) {
+              return null
+            }
+            const loaded = await ctx.loadDoc(id)
+            if (loaded == null) {
+              return null
+            }
+            const loadedModel = loaded.metadata.model.toString()
+            if (relation.model != null && loadedModel !== relation.model) {
+              console.warn(
```

Comment:
When can this happen?

Reply:
That could happen if the server doesn't perform this check when creating or querying relations, this ensures it doesn't create a situation the client couldn't handle.